libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef.dat

```diff
@@ -0,0 +1,159 @@
+# hw definition file for processing by chibios_pins.py
+# for Holybro KakuteH7 hardware.
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1048
+
+# crystal frequency, setup to use external oscillator
+OSCILLATOR_HZ 8000000
+
+FLASH_SIZE_KB 2048
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 12
+define CH_CFG_ST_RESOLUTION 16
+
+# leave 1 sectors free
+FLASH_RESERVE_START_KB 128
+
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+define STORAGE_FLASH_PAGE 14
+define HAL_STORAGE_SIZE 32768
+
+# one I2C bus
+I2C_ORDER I2C1
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 UART7 USART6 OTG2
+
+# buzzer
+PC13 BUZZER OUTPUT LOW PULLDOWN GPIO(80)
+define HAL_BUZZER_PIN 80
+define HAL_BUZZER_ON 1
+define HAL_BUZZER_OFF 0
+
+# PA10 IO-debug-console
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 for SDCard
+PA4 SDCARD_CS CS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+PA3 SDCARD_DETECT INPUT
+
+# SPI2 for MAX7456 OSD
+PB12 MAX7456_CS CS
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# SPI4 for ICM20689
+PE4 ICM20689_CS CS
+PE2 SPI4_SCK SPI4
+PE5 SPI4_MISO SPI4
+PE6 SPI4_MOSI SPI4
+
+# I2C1 for baro
+PB6 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+PA8 VBUS INPUT
+
+# define default battery setup
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 10.1
+define HAL_BATT_CURR_SCALE 17.0
+
+PC5 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 8
+
+PC2 LED0 OUTPUT LOW
+
+# USART1
+PA9  USART1_TX USART1
+PA10 USART1_RX USART1
+
+# USART2
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+
+# USART3 (GPS)
+PD8 USART3_TX USART3
+PD9 USART3_RX USART3
+
+# UART4 (GPS2)
+PD0 UART4_RX UART4 NODMA
+PD1 UART4_TX UART4 NODMA
+
+# RC input defaults to timer capture
+PC7 TIM8_CH2 TIM8 RCININT PULLDOWN
+
+# alternative config using USART for protocols
+# like FPort
+PC7 USART6_RX USART6 NODMA ALT(1)
+PC6 USART6_TX USART6 NODMA
+
+# UART7 USED BY ESC FROM ORIGINAL DESIGN
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+
+# Motors
+PB0  TIM1_CH2N TIM1 PWM(1) GPIO(50)
+PB1  TIM1_CH3N TIM1 PWM(2) GPIO(51)
+PB3  TIM2_CH2  TIM2 PWM(3) GPIO(52)
+PB10 TIM2_CH3  TIM2 PWM(4) GPIO(53)
+PA0  TIM5_CH1  TIM5 PWM(5) GPIO(54)
+PA2  TIM5_CH3  TIM5 PWM(6) GPIO(55)
+PC8  TIM3_CH3  TIM3 PWM(7) GPIO(56)
+PC9  TIM3_CH4  TIM3 PWM(8) GPIO(57)
+
+# LED strip
+PD12 TIM4_CH1 TIM4 PWM(9) GPIO(58)
```

Comment:
I think it helps to include a defaults.parm file that setups the LED pin as NeoPixel

Reply:
done